lib/jekyll/geolexica/bibliography_generator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# (c) Copyright 2020 Ribose Inc.
+#
+
+module Jekyll
+  module Geolexica
+    class BibliographyGenerator < Generator
```

Comment:
Not sure why we need to use Jekyll::Generator?

Ideally I want to separate our code from Jekyll -- Jekyll is just used to generate the pages, but the concepts pre-processing code should really be separate (a separate root module that is not `module Jekyll`)

Can we start a `module Geolexica` and keep the concepts processing code all there?

Reply:
@ronaldtse  Currently, there is no way to run a script before the site generation here in `geolexica-server` so that is why I added the code for bibliography generation in a generator.
It was running using the make file in [isotc-204.geolexica.org](https://github.com/geolexica/isotc204.geolexica.org/blob/main/Makefile).
Even if I move the code to `module Geolexica` I am a bit confused about how will it run before the site generation.

Reply:
Ultimately jekyll is just one expression medium for the concept dataset. 

I am more considering a separate rake task that prepares the data (pre-processes) for Jekyll using the `Geolexica` module, and then run Jekyll (we really only use its two commands, `build` and `serve`) to build the site.

Maybe we are looking at `geolexica serve` and `geolexica build` to wrap around these processes around Jekyll.

Reply:
@ronaldtse Should we move this to a new gem or add the module in this repo?

If you prefer to move this to a new gem than what do you suggest we should name that gem?

Reply:
@HassanAkbar actually this repository was meant as `geolexica-server`, it's just that we also have the `jekyll-geolexica` gem within. I would suggest we continue with this repository, and them move the jekyll-geolexica gem out to a new repository.

Reply:
I've created the jekyll-geolexica repository that we can move things over (https://github.com/geolexica/jekyll-geolexica)